akka-stream/src/main/scala/akka/stream/ThrottleMode.scala

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream
+
+/**
+ * Represents a mode that decides how to deal exceed rate for Throttle combinator
+ */
+sealed abstract class ThrottleMode
+
+object ThrottleMode {
+
+  /**
+   *  Tells throttle to make pauses before emitting messages to meet throttle rate
+   */
+  case object ShapingMode extends ThrottleMode
```

Comment:
since it is in object `ThrottleMode` I would only call them `Shaping` and `Enforcing`